dali/kernels/audio/mel_scale/mel_filter_bank_gpu_test.cc

```diff
@@ -152,28 +152,166 @@ TEST_P(MelScaleGpuTest, MelScaleGpuTest) {
+INSTANTIATE_TEST_SUITE_P(MelScaleGpuTest, MelScaleGpuTest, testing::Combine(
+    testing::Values(std::vector<TensorShape<>>{TensorShape<>{10, 4, 6, 65}},
+                    std::vector<TensorShape<>>{TensorShape<>{4, 5, 6, 17},
+                                               TensorShape<>{4, 8, 6, 17}}),  // shape
     testing::Values(4, 8),  // nfilter
     testing::Values(16000.0f),  // sample rate
     testing::Values(0.0f, 1000.0f),  // fmin
-    testing::Values(5000.0f, 8000.0f),  // fmax
+    testing::Values(8000.0f, 5000.0f),  // fmax
     testing::Values(0, 2, 3)));  // axis
 
+using BenchBase = ::testing::TestWithParam<std::tuple<
+    int,                  // axis
+    int,                  // channels
+    std::tuple<int, int, int>,  // batch_size, min_length, max_length
+    int,                  // nfft
+    int,                  // nfilter
+    float,                // sample_rate
+    float,                // fmin
+    float>>;              // fmax
+
+class MelScaleGpuBench : public BenchBase {
+ public:
+  MelScaleGpuBench() {
+    std::tuple<int, int, int> data_sizes;
+    std::tie(
+      axis_,
+      channels_,
+      data_sizes,
+      nfft_,
+      nfilter_,
+      sample_rate_,
+      freq_low_,
+      freq_high_) = GetParam();
+    std::tie(batch_size_, min_length_, max_length_) = data_sizes;
+    assert(axis_ == 0 || axis_ == 1);
+  }
+
+ protected:
+  void SetUp() final {
+    std::mt19937 rng(12345);
+    in_shape_.resize(batch_size_, channels_ > 1 ? 3 : 2);
+    std::uniform_int_distribution<int> length_dist(min_length_, max_length_);
+    int length_axis = 1 - axis_;
+    for (int i = 0; i < batch_size_; i++) {
+      auto sh = in_shape_.tensor_shape_span(i);
+      sh[axis_] = nfft_;
+      sh[length_axis] = length_dist(rng);
+      if (channels_ > 1)
+        sh[2] = channels_;
+    }
+    out_shape_ = in_shape_;
+    for (int i = 0; i < batch_size_; ++i) {
+      out_shape_.tensor_shape_span(i)[axis_] = nfilter_;
+    }
+
+    in_.reshape(in_shape_);
+    auto in_view = in_.gpu();
+    for (int i = 0; i < in_view.num_samples(); i++)
+      CUDA_CALL(cudaMemset(in_view.data[i], rng(), in_view[i].num_elements() * sizeof(float)));
+  }
+  int axis_ = 0;
+  int channels_ = 1;
+  int batch_size_ = 16;
+  int min_length_ = 1000;
+  int max_length_ = 10000;
+  int nfft_ = 257;
+  int nfilter_ = 80;
+  float sample_rate_ = 16000, freq_low_ = 0, freq_high_ = 8000;
+  TensorListShape<> in_shape_, out_shape_;
+  TestTensorList<float> in_;
+};
+
+TEST_P(MelScaleGpuBench, Benchmark) {
+  auto stream = CUDAStreamPool::instance().Get();
+
+  KernelContext ctx;
+  ctx.gpu.stream = stream;
+  kernels::audio::MelFilterBankArgs args;
+  kernels::KernelManager kmgr;
+  args.axis = axis_;
+  args.nfft = nfft_;
+  args.nfilter = nfilter_;
+  args.sample_rate = sample_rate_;
+  args.freq_low = freq_low_;
+  args.freq_high = freq_high_;
+  args.mel_formula = MelScaleFormula::Slaney;
+  args.normalize = false;
+
+  print(std::cout,
```

Comment:
why not `std::cout <<` ?

Reply:
1. Shorter
2. Automatically uses to_string, when applicable